simtools/visualization/visualize.py

```diff
@@ -531,3 +542,141 @@ def plot_hist_2D(data, **kwargs):
+            width=radius.value,
+            height=radius.value,
+            fill=False,
+            color=colors_dict[sct],
+        )
+    else:
+        patch = mpatches.Circle(
+            (x.value, y.value),
+            radius=radius.value,
+            fill=fill_flag,
+            color=colors_dict[valid_name],
+        )
+    return patch
+
+
+@u.quantity_input(rotation_angle=u.deg)
+def plot_array(telescopes, rotate_angle=0 * u.deg):
+    """
+    Plot the array of telescopes.
+
+    Parameters
+    ----------
+    telescopes: dict
+        Dictionary with the telescope position and names.
+    rotate_angle:
+        Angle to rotate the plot. For rotate_angle = 0 the x-axis points towards the east, and\
+        the y-axis points towards the North.
+
+    Returns
+    -------
+    plt.figure
+        Instance of plt.figure with the array of telescopes plotted.
+
+    """
+
+    fig, ax = plt.subplots(1)
+    legend_objects = []
+    legend_labels = []
+    tel_counters = {one_telescope: 0 for one_telescope in names.all_telescope_class_names}
+    if rotate_angle != 0:
+        telescopes["pos_x"], telescopes["pos_y"] = gen.rotate(
+            rotate_angle, telescopes["pos_x"], telescopes["pos_y"]
+        )
+
+    size_factor = max(np.max(telescopes["pos_x"]), np.max(telescopes["pos_y"])) / (300.0 * u.m)
+    fontsize = 12
+
+    patches = []
+    for i_tel, tel_now in enumerate(telescopes):
+        for tel_type in tel_counters:
+            if tel_type in tel_now["telescope_name"]:
+                tel_counters[tel_type] += 1
+        i_tel_name = telescopes[i_tel]["telescope_name"][:3]
+        if i_tel_name == sst:
+            fontsize = 5
+        patches.append(
+            get_telescope_patch(
+                i_tel_name,
+                telescopes[i_tel]["pos_x"],
+                telescopes[i_tel]["pos_y"],
+                telescopes[i_tel]["radius"] * size_factor,
+            )
+        )
+        ax.text(
+            telescopes[i_tel]["pos_x"].value,
+            telescopes[i_tel]["pos_y"].value + telescopes[i_tel]["radius"].value,
+            telescopes[i_tel]["telescope_name"],
+            horizontalalignment="center",
+            verticalalignment="bottom",
+            fontsize=fontsize,
+        )
+
+    for step, one_telescope in enumerate(names.all_telescope_class_names):
+        if tel_counters[one_telescope] > 0:
+            legend_objects.append(leg_h.all_telescope_objects[step]())
+            legend_labels.append(one_telescope + f" ({tel_counters[one_telescope]})")
+
+    plt.gca().add_collection(PatchCollection(patches, match_original=True))
+
+    x_title = "East [m]"
+    y_title = "North [m]"
+    plt.axis("square")
+    plt.grid(True)
+    plt.gca().set_axisbelow(True)
+    plt.xlabel(x_title, fontsize=18, labelpad=0)
+    plt.ylabel(y_title, fontsize=18, labelpad=0)
+    plt.tick_params(axis="both", which="major", labelsize=15)
+
+    legend_handler_map = {
```

Comment:
I just remember the reason. it is because here we are initializing the handlers. by defining this in the legend_handlers, every time we import leg_h the handlers will be initialized and this leads to an error. I dont see a problem in defining it here where it is actually used

Reply:
Oh right, then it's perfectly fine. What about the question whether you can write it explicitly instead of building the map from the lists?
(I admit I did not recheck if that question is still relevant considering your answer, so if it isn't, please just say that it isn't relevant anymore.)

Reply:
We could write it explicitly but that would mean we would be hard-wiring the telescope handlers. with the current implementation we get the list the list with all the handlers defined in legend_handlers. we would have to make sure to keep that one list updated but we dont need to worry about the use of the handlers elsewhere.